ethcore/src/spec/spec.rs

```diff
@@ -226,59 +296,51 @@ fn load_from(spec_params: SpecParams, s: ethjson::spec::Spec) -> Result<Spec, Er
 	let GenericSeal(seal_rlp) = g.seal.into();
 	let params = CommonParams::from(s.params);
 
-	let hardcoded_sync = if let Some(ref hs) = s.hardcoded_sync {
-		if let Ok(header) = hs.header.from_hex() {
-			Some(SpecHardcodedSync {
-				header: encoded::Header::new(header),
-				total_difficulty: hs.total_difficulty.into(),
-				chts: s.hardcoded_sync
-					.as_ref()
-					.map(|s| s.chts.iter().map(|c| c.clone().into()).collect())
-					.unwrap_or_default()
-			})
-		} else {
-			None
-		}
-	} else {
-		None
-	};
+	let hardcoded_sync = s.hardcoded_sync.map(Into::into);
 
-	let mut s = Spec {
+	let engine = Spec::engine(spec_params, s.engine, params, builtins);
+	let author = g.author;
+	let timestamp = g.timestamp;
+	let difficulty = g.difficulty;
+	let constructors: Vec<_> = s.accounts
+		.constructors()
+		.into_iter()
+		.map(|(a, c)| (a.into(), c.into()))
+		.collect();
+	let genesis_state: PodState = s.accounts.into();
+
+	let (state_root_memo, _) = run_constructors(
```

Comment:
by running constructors always before `Spec` is created, we ensure that `spec_root_memo` is always correct